lib/drivers/medtronic600/medtronic600Simulator.js

```diff
@@ -482,16 +482,29 @@ class Medtronic600Simulator {
 
       if (suppressedBasal != null) {
         if (Date.parse(suppressedBasal.basalEndTime) > Date.parse(activeSuspendedBasal.time)) {
-          suppressedBasal.duration = Date.parse(activeSuspendedBasal.time) -
-            Date.parse(suppressedBasal.time);
-          // It's possible for duration to get down to 0 here (which is fine).
-          if (suppressedBasal.duration < 0) {
-            throw new Error('Should never get a duration of less than zero');
-          } else if (suppressedBasal.duration === 0) {
+          if (Date.parse(activeSuspendedBasal.time) >= Date.parse(suppressedBasal.time)) {
+            suppressedBasal.duration = Date.parse(activeSuspendedBasal.time) -
+              Date.parse(suppressedBasal.time);
+          } else {
+            // The suspend event is the first basal event in the history.
+            // Because we don't know what it's suspending, match it to the beginning of the
+            // first suspended event.
+            activeSuspendedBasal.time = suppressedBasal.time;
```

Comment:
it's usually dangerous to set `.time` without using `fillInUTCInfo`. Do we have to set `.time` here manually?

Reply:
This is a unique case of us modifying the suspend event from its original. In this instance, we pretend that the event occurred after it really did (to the beginning of the first known basal event, which is already "UTC happy"). We do this because we have no idea of the basal event that the suspend is "suspending". 

Thinking through it further, though - this may be causing an issue for the deduplicator though, since it will look like a different event. I'm wondering if we should discard the entire suspended event?

Reply:
Which fields are changing here to cause this to look like a different event? Only some fields are used in the deduplication process and those fields depend somewhat on the type of event.

Reply:
I would suggest one of two options:

1. Keep it's original time, but set the duration to 0 and annotate with `basal/unknown-duration`
1. Drop the event, since we don't really have enough information to create a record (as in, both `duration` and `suppressed` is missing) and there's a small chance option 1  could overwrite the event from a previous upload, depending on how we perform deduplication (either now or in the future)

Reply:
@darinkrauss - the start date/time is changing.
@gniezen - in either case, we'd also need to ignore the underlying supressed basals as well. In case 1, we'd need to ignore the basals that _would_ have been suppressed by the actual duration, and in case 2, we'd need to drop the events that happen during the suspend period.

Reply:
> We do this because we have no idea of the basal event that the suspend is "suspending".

In that case, should we not keep the time of the suspended basal event as is, and just not fill in the suppressed field? (As from the Uploader's PoV, we don't have it)

Reply:
I think I have to do some testing to see how the deduplicator handles things...